runtime/parachains/src/session_info.rs

```diff
@@ -259,23 +263,29 @@ mod tests {
 
 	#[test]
 	fn session_pruning_avoids_heavy_loop() {
-		new_test_ext(genesis_config()).execute_with(|| {
-			let start = 1_000_000_000;
-			System::on_initialize(start);
-			System::set_block_number(start);
-
-			if let Some(notification) = new_session_every_block(start) {
-				Configuration::initializer_on_new_session(&notification.validators, &notification.queued);
-				SessionInfo::initializer_on_new_session(&notification);
-			}
-
-			Configuration::initializer_initialize(start);
-			SessionInfo::initializer_initialize(start);
-
-			assert_eq!(EarliestStoredSession::get(), start - 1);
-
-			run_to_block(start + 1, new_session_every_block);
-			assert_eq!(EarliestStoredSession::get(), start);
+		futures::executor::block_on(async {
+			async {
+				new_test_ext(genesis_config()).execute_with(|| {
+					let start = 1_000_000_000;
+					System::on_initialize(start);
+					System::set_block_number(start);
+
+					if let Some(notification) = new_session_every_block(start) {
+						Configuration::initializer_on_new_session(&notification.validators, &notification.queued);
+						SessionInfo::initializer_on_new_session(&notification);
+					}
+
+					Configuration::initializer_initialize(start);
+					SessionInfo::initializer_initialize(start);
+
+					assert_eq!(EarliestStoredSession::get(), start);
+
+					run_to_block(start + 1, new_session_every_block);
+					assert_eq!(EarliestStoredSession::get(), start);
+				})
+			}.timeout(Duration::from_secs(5))
```

Comment:
This is a really bad idea. Never use such timeouts in stuff that runs in CI

Reply:
This is not the first place we use such timeouts, just search for `timeout`.
Do you have an alternative to what this test is trying to do?

Reply:
You could just insert a Session at 500000 or whatever and check that this isn't removed. Before it should have been removed and with your fixes this should not happen anymore.

Reply:
Ok, I'll change it, but the point of the test is not about deletion, but rather it's not taking too long, which is what is tested by timeout.
https://github.com/paritytech/polkadot/pull/2099 should provide more context.

Reply:
I read this. But if you see that the test is removing something that should not be removed or if it takes to long or whatever. You know that something is broken.

An infinite running test is also good, because you still know that you broke something.